src/audio/volume/volume.c

```diff
@@ -1217,6 +1215,39 @@ static vol_zc_func vol_get_zc_function(struct comp_dev *dev)
 	return NULL;
 }
 
+/**
+ * \brief Set volume frames alignment limit.
+ * \param[in,out] source Structure pointer of source.
+ * \param[in,out] sink Structure pointer of sink.
+ */
+static void volume_set_alignment(struct audio_stream *source,
+				 struct audio_stream *sink)
+{
+#if XCHAL_HAVE_HIFI3 || XCHAL_HAVE_HIFI4
+
+	/* both source and sink buffer in HiFi 3 or HiFi4 processing version,
+	 * xtensa intrinsics ask for 8-byte aligned. 5.1 format SSE audio
+	 * requires 16-byte aligned.
+	 */
+	const uint32_t byte_align = source->channels == 6 ? 16 : 8;
+
+	/*there is no limit for frame number, so both source and sink set it to be 1*/
+	uint32_t frame_align_req = 1;
+
+#else
+
+	/* Since the generic version process signal sample by sample, so there is no
+	 * limit for it, then set the byte_align and frame_align_req to be 1.
+	 */
+	uint32_t byte_align = 1;
+	uint32_t frame_align_req = 1;
```

Comment:
```suggestion
	const uint32_t frame_align_req = 1;
```
Same with `byte_align`. Use `const` everywhere you can (in some modern languages `const` is the default)

Is there some reason for `frame_align_req` to be inside the `#if #else`?